libraries/chain/include/eosio/chain/fork_database.hpp

```diff
@@ -178,11 +184,11 @@ namespace eosio::chain {
       template <class R, class F>
       R apply_s(const F& f) {
          if constexpr (std::is_same_v<void, R>) {
-            if (!legacy) {
+            if (in_use == in_use_t::savanna || in_use == in_use_t::both) {
```

Comment:
This need to be something like `auto in_use_value = in_use; if (in_use_value == in_use_t::savanna || in_use_value == in_use_t::both) {` as the value can change between the `||`. Same for the rest of the checks below.

Reply:
Updated